c7n/resources/s3.py

```diff
@@ -1372,6 +1372,144 @@ def process_bucket(self, bucket):
+                    scope: Any
+                    state: absent
+    """
+
+    schema = type_schema(
+        'check-public-block',
+        scope={
+            'type': 'string',
+            'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
+                'BlockPublicPolicy', 'RestrictPublicBuckets', 'All', 'Any']},
+        state={
+            'type': 'string',
+            'enum': ['absent', 'present']},
+        required=['scope', 'state'])
+    permissions = ("s3:GetBucketPublicAccessBlock",)
+
+    def process(self, buckets, event=None):
+        results = list()
+        with self.executor_factory(max_workers=2) as w:
+            futures = {w.submit(self.process_bucket, bucket): bucket for bucket in buckets}
+            for future in as_completed(futures):
+                bucket = futures[future]
+                if future.exception():
+                    continue
+                if future.result():
+                    results.append(bucket)
+        return results
+
+    def process_bucket(self, bucket):
+        s3 = bucket_client(local_session(self.manager.session_factory), bucket)
+        state = self.data.get('state')
+        scope = self.data.get('scope')
+        try:
+            config = s3.get_public_access_block(
+                Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                config = None
+            else:
+                raise
+        if config:
+            if scope == 'All':
+                return all(config.values()) if state == 'present' else not any(config.values())
+            elif scope == 'Any':
+                return any(config.values()) if state == 'present' else not all(config.values())
+            else:
+                return config[scope] if state == 'present' else not config[scope]
+        else:
+            # return true for a null config, meaning no public blocks
+            return True if state == 'absent' else False
+
+
+@actions.register('set-public-block')
+class SetPublicBlock(BucketActionBase):
+    """Action to update Public Access blocks on S3 buckets
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: s3-dont-ignore-public-acls
+                resource: s3
+                filters:
+                  - type: check-public-block
+                    scope: Any
+                    state: absent
+                actions:
+                  - type: set-public-block
+                    scope: All
+                    state: enable
+    """
+
+    schema = type_schema(
+        'set-public-block',
+        scope={
+            'type': 'string',
+            'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
+                'BlockPublicPolicy', 'RestrictPublicBuckets', 'All']},
+        state={'type': 'string',
+               'enum': ['enable', 'disable']},
+        required=['scope', 'state'])
```

Comment:
I would argue that the defaults should be `All` and `enable` and that they shouldn't be required settings...

Reply:
This makes sense to me too. Will add in the next PR.

Reply:
Resolved in 86996ae5b7488f5023564af4f205c8550c14378c